File: lib/MESCIUS/ComponentOne/WinForms/C1TouchToolKit/PolicySourceCodes/WPF_ElementHostZoomPolicy.cs
using System;
using System.Collections.Generic;
using System.Text;
using System.Windows.Forms.Integration;
using System.Windows.Media;
using GrapeCity.Win.MultiTouch;
using System.Windows;

namespace GrapeCity.Win.MultiTouch.ZoomPolicys
{
    public class WPF_ElementHostZoomPolicy : NoZoomFontZoomPolicy
    {
        public override Type TargetType
        {
            get { return typeof(ElementHost); }
        }

        public override bool CanZoomChildren(System.Windows.Forms.Control control)
        {
            return false;
        }

        public override void ZoomBounds(System.Windows.Forms.Control control, ZoomBoundsInfo infos)
        {
            ElementHost host = control as ElementHost;
            if (host.Child is FrameworkElement)
            {
                (host.Child as FrameworkElement).LayoutTransform = new ScaleTransform(infos.TargetFactor, infos.TargetFactor);
            }
            base.ZoomBounds(control, infos);
        }
    }
}

File: lib/MESCIUS/ComponentOne/WinForms/C1TouchToolKit/PolicySourceCodes/WPF_ElementHostZoomPolicy.vb
Imports System.Collections.Generic
Imports System.Text
Imports System.Windows.Forms.Integration
Imports System.Windows.Media
Imports GrapeCity.Win.MultiTouch
Imports System.Windows

Public Class WPF_ElementHostZoomPolicy
	Inherits NoZoomFontZoomPolicy
	Public Overrides ReadOnly Property TargetType() As Type
		Get
			Return GetType(ElementHost)
		End Get
	End Property

	Public Overrides Function CanZoomChildren(control As System.Windows.Forms.Control) As Boolean
		Return False
	End Function

	Public Overrides Sub ZoomBounds(control As System.Windows.Forms.Control, infos As ZoomBoundsInfo)
		Dim host As ElementHost = TryCast(control, ElementHost)
		If TypeOf host.Child Is FrameworkElement Then
			TryCast(host.Child, FrameworkElement).LayoutTransform = New ScaleTransform(infos.TargetFactor, infos.TargetFactor)
		End If
		MyBase.ZoomBounds(control, infos)
	End Sub
End Class
